pubsub/pubsub_test.go

```diff
@@ -0,0 +1,201 @@
+// Copyright 2018 The Go Cloud Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package pubsub_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/google/go-cloud/pubsub"
+	"github.com/google/go-cloud/pubsub/driver"
+)
+
+type driverTopic struct {
+	subs []*driverSub
+}
+
+func (t *driverTopic) SendBatch(ctx context.Context, ms []*driver.Message) error {
+	for _, s := range t.subs {
+		select {
+		case <-s.sem:
+			s.q = append(s.q, ms...)
+			s.sem <- struct{}{}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+	return nil
+}
+
+func (t *driverTopic) Close() error {
+	return nil
+}
+
+type driverSub struct {
+	sem chan struct{}
```

Comment:
I think you could replace these two with a buffered channel.

Reply:
(Thinking ahead to when I steal this for the fake:) But what is the right buffer size? We should support publishing an arbitrary amount of messages before any are received, as the backend would.